src/ring_theory/dedekind_domain.lean

```diff
@@ -0,0 +1,43 @@
+import ring_theory.fractional_ideal
+import ring_theory.ideal.over
+import ring_theory.discrete_valuation_ring
+
+/-- A ring `R` is (at most) one-dimensional if all nonzero prime ideals are maximal. -/
+def ring.is_one_dimensional (R : Type*) [comm_ring R] :=
+∀ p ≠ (⊥ : ideal R), p.is_prime → p.is_maximal
+
+open ideal ring
+
+lemma principal_ideal_ring.is_one_dimensional (R : Type*) [integral_domain R]
+  [is_principal_ideal_ring R] :
+  ring.is_one_dimensional R :=
+λ p nonzero prime, by { haveI := prime, exact is_prime.to_maximal_ideal nonzero }
+
+variables {R K : Type*}
+
+lemma integral_closure.is_one_dimensional [comm_ring R] [nontrivial R] [integral_domain K] [algebra R K]
+  (h : is_one_dimensional R) :
+  is_one_dimensional (integral_closure R K) :=
+begin
+  intros p ne_bot prime,
+  haveI := prime,
+  refine integral_closure.is_maximal_of_is_maximal_comap p (h _ (integral_closure.comap_ne_bot ne_bot) _),
+  apply is_prime.comap
+end
+
+/--Noetherian, integrally closed, Krull dimension 1-/
```

Comment:
This docstring doesn't make sense when someone reads it in a pop-up outside of this file. It would be great if this docstring explained what the *definition* of `is_dedekind_domain` is (you've tersely done that, but you should mention the words "Dedekind domain"), but also use this opportunity to mention the other definitions (and announce that it is still a TODO to prove that they are equivalent).

Reply:
(We also generally have single spaces after `/--` and before `-/`.)